crates/bevy_asset/src/server/mod.rs

```diff
@@ -1339,6 +1339,28 @@ pub enum LoadState {
     Failed(Box<AssetLoadError>),
 }
 
+impl LoadState {
+    /// Returns `true` if this instance is [`LoadState::Loading`]
+    pub fn is_loading(&self) -> bool {
+        matches!(self, Self::Loading)
+    }
+    /// Returns `true` if this instance is [`LoadState::Loaded`]
+    pub fn is_loaded(&self) -> bool {
+        matches!(self, Self::Loaded)
+    }
+    // NOTE: an `is_not_loaded` method is intentionally not included, as it may mislead some users
+    // into thinking it is complementary to `is_loaded`.
+    // `NotLoaded` is a very specific failure mode and in most cases it is not necessary to directly check for it.
+    // If this is necessary the `matches!` macro can be used instead of a helper method.
+
+    /// Returns `true` if the asset is loaded or in the process of being loaded. If true true,
+    /// then the asset can be considered to be in a "normal" state: the asset either exists
+    /// or will exist, and no errors have been encountered yet.
+    pub fn is_loaded_or_loading(&self) -> bool {
+        self.is_loaded() || self.is_loading()
+    }
```

Comment:
Is this a common thing yo check for? I imagine you'd generally want to handle these two cases separately, right?

Reply:
I'm not tied to this method, I'd be okay with removing it. I just imagined that some users would want a quick way of checking if an asset is in a valid state or not, and allowing you to call `is_loaded` and `is_loading` in a single call allows you to avoid binding a `LoadState` to a local